spec/message.abnf

```diff
@@ -47,12 +49,15 @@ unquoted-start = name-start / DIGIT / "."
 reserved       = ( reserved-start / private-start ) reserved-body
 reserved-start = "!" / "@" / "#" / "%" / "*" / "<" / ">" / "/" / "?" / "~"
 private-start  = "^" / "&"
-reserved-body  = *( [s] 1*(reserved-char / reserved-escape / literal))
+reserved-body  = *( [s] 1*(reserved-char / reserved-escape / quoted))
```

Comment:
This is an unrelated fix that I noticed while working on this PR.

Reply:
**Please make a separate PR for this.**

The intention was to allow `reserved` to define whatever structure it wants. For example, allowing "comments" might look something like:

```
{% This is a "comment". Do not translate it}
{$foo % This is a "comment" about $foo: do not translate it}
```

The contents of  `reserved` must appear within a placeholder, so the characters `{` and `}` must be escaped--but everything else (including quotes, whichever ones we use) remain undefined.

Hence `reserved-char` should match `text-char`. The production `reserved-char` currently omits whitespace, but the optional whitespace part of `reserved-body` just adds the spaces back.

Reply:
Done: #415.